### DISCUSSION TITLE
Is there a way to generate the builder blocks based on state changes?

### DISCUSSION THREAD

**Discussion by aymanalhattami** [filamentphp/filament#10015] [Help]
I need to implement the idea in this video (product variations) https://www.youtube.com/watch?v=njdHCpildhE with filament.

The best choices are builder or repeater, but I need to change the fields of repeater or blocks of builder based on state changes. and I did not find any way to do that.

**Comment by remiebelingmerifond** [comment]
Just in case someone else ends up here searching for a solution, like I did:
I had a similar problem and came up with this:

```php
 Forms\Components\Select::make('type')
    ->label('Product Type')
    ->options([
        'Option a',
        'Option b'
    ])
    ->live(),
 Forms\Components\Builder::make('Foo')
     ->blocks(fn ($get) => SomeSmartBlocksCollectingClass::getBuilderBlocksByType($get('type')))      
```

Of course, blocks can also be defined inline.

It should look something like this:

**Careful! The following is untested pseudo-code:**

```php
Forms\Components\Select::make('type')
    ->label('Product Type')
    ->options([
        'Option a',
        'Option b'
    ])
    ->live(),
 Forms\Components\Builder::make('Foo')
     ->blocks(function ($get) {
        return match($get('type')) {
           1 => [/* Array  of Block definitions */],
           2 => [/* Array  of Block definitions */],
           default => [/* Array  of Block definitions */],
        };
}) 
```
